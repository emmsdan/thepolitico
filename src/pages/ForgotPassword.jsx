import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import CallToAction from '../components/presentationals/CallToAction/CallToAction';
// import PropTypes from 'prop-types'
import FormCard from '../components/presentationals/Form/FormCard';
import FormInput from '../components/presentationals/Form/FormInput';
import FormButton from '../components/presentationals/Form/FormButton';
import { generateFormData } from '../utils/form';
import { authenticationRequestHandler } from '../store/reducers/auth';
import Loader from '../components/presentationals/Loader/Loader';
import { authenticationToken } from '../utils/helpers';

class ForgotPassowrd extends Component {
  componentDidMount() {
    const { history } = this.props;
    authenticationToken() && history.push('/');
  }

  submitResetPasswordDetails = event => {
    event.preventDefault();
    const formData = generateFormData(event.target);
    return this.props.authenticationRequestHandler(formData, 'reset', [], '/');
  };

  render() {
    return (
      <>
        {this.props.auth.isLoading && (
          <Loader loader="show" text="Hi, am processing your input" />
        )}
        <CallToAction withButton={false} />
        <section className="wrapper">
          <FormCard
            header="Sign in to Continue"
            onSubmit={this.submitResetPasswordDetails}
          >
            <FormInput
              name="email"
              id="email"
              placeholder="enter email address"
              title="Email"
            />
            <br />
            <FormButton type="submit" text="Login" />
          </FormCard>
        </section>
      </>
    );
  }
}

ForgotPassowrd.propTypes = {
  auth: PropTypes.object,
  url: PropTypes.string,
  poweredby: PropTypes.shape({
    url: PropTypes.string,
    name: PropTypes.string,
  }),
  authenticationRequestHandler: PropTypes.func.isRequired,
  location: PropTypes.object,
  history: PropTypes.object,
};

const mapStateToProps = state => {
  return { auth: state.auth };
};

export default connect(
  mapStateToProps,
  { authenticationRequestHandler },
)(ForgotPassowrd);
