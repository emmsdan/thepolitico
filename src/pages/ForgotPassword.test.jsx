import React from 'react';
import { mount } from 'enzyme';

import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import ForgotPassword from './ForgotPassword';
let store = {};
describe('ForgotPassword', () => {
  beforeEach(() => {
    const mockStore = configureMockStore();
    store = mockStore({
      history: [],
      location: {},
      auth: { isLoading: true },
      authenticationRequestHandler: jest.fn(),
    });
  });

  it('should render correctly', () => {
    const component = mount(<ForgotPassword store={store} />);
    expect(component).toMatchSnapshot();
  });

  it('should simulate submit correctly', () => {
    const component = mount(
      <Provider store={store}>
        <ForgotPassword />
      </Provider>,
    );
    const node = component.find('button').simulate('click');
    expect(node).toMatchSnapshot();
  });
});
