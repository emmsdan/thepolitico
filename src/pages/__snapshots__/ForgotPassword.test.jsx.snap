// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPassword should render correctly 1`] = `
<ConnectFunction
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ForgotPassowrd
    auth={
      Object {
        "isLoading": true,
      }
    }
    authenticationRequestHandler={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Loader
      loader="show"
      text="Hi, am processing your input"
    >
      <div
        className="center load-overlay show"
      >
        <div>
          <div
            className="loader center"
            style={
              Object {
                "backgroundImage": "url([object Object])",
                "backgroundRepeat": "no-repeat",
                "backgroundSize": "contain",
                "borderRadius": 0,
              }
            }
          />
          <strong>
            Hi, am processing your input
            ...
          </strong>
        </div>
      </div>
    </Loader>
    <CallToAction
      withButton={false}
    >
      <section
        className="call-to-action"
      >
        <div
          className="placeholder"
        >
          <h2>
            Every Vote Count
          </h2>
          <h4>
            Vote and Be Voted for
          </h4>
        </div>
      </section>
    </CallToAction>
    <section
      className="wrapper"
    >
      <FormCard
        header="Sign in to Continue"
        onSubmit={[Function]}
      >
        <form
          className="form-fields"
          header="Sign in to Continue"
          onSubmit={[Function]}
        >
          <fieldset
            className="fieldset"
          >
            <legend>
              Sign in to Continue
            </legend>
            <FormInput
              id="email"
              name="email"
              placeholder="enter email address"
              title="Email"
            >
              <div
                className="field"
              >
                <span
                  className="error red-text"
                  htmlFor="email"
                />
                <input
                  id="email"
                  name="email"
                  placeholder="enter email address"
                  title="Email"
                />
                <label
                  htmlFor="email"
                >
                  Email
                  :
                </label>
              </div>
            </FormInput>
            <br />
            <div
              className="field"
            >
              <button
                className="button-bigger"
                type="submit"
              >
                Reset Password
              </button>
            </div>
          </fieldset>
        </form>
      </FormCard>
    </section>
  </ForgotPassowrd>
</ConnectFunction>
`;

exports[`ForgotPassword should simulate submit correctly 1`] = `
<button
  className="button-bigger"
  type="submit"
>
  Reset Password
</button>
`;
